golang/thrift/outbound.go

```diff
@@ -0,0 +1,323 @@
+package thrift
+
+import "bytes"
+import "errors"
+import "github.com/apache/thrift/lib/go/thrift"
+import tchannel "github.com/uber/tchannel/golang"
+import "golang.org/x/net/context"
+import "time"
+
+// NewTChannelOutboundProtocol creates a TChannelOutboundProtocol
+func NewTChannelOutboundProtocol(remoteHostPort, remoteServiceName, remoteProcessorName string) (*TChannelOutboundProtocol, error) {
+	tchannel, err := tchannel.NewChannel("0.0.0.0:0", nil)
```

Comment:
We should pass the actual TChannel object into this function, so that we can re-use connections across calls.